src/classes/QueryParentPredicate.cls

```diff
@@ -0,0 +1,31 @@
+public class QueryParentPredicate extends QueryPredicate {
```

Comment:
this is the addition for querying parent fields.  Because our implementation of QueryBuilder and SObjectRepository are already relying heavily on a generic "get" method by List<QueryPredicate>, pushing the implementation to the predicate felt right.  

The utility of the predicate really has me feeling like I would actually prefer to rip out the other get methods in the interface and rely fully on the predicate.